kanidm_client/src/asynchronous.rs

```diff
@@ -96,13 +96,13 @@ impl KanidmAsyncClient {
             let mut sguard = self.auth_session_id.write().await;
             *sguard = headers
                 .get(KSESSIONID)
-                .map(|hv| hv.to_str().ok().map(|s| s.to_string()))
+                .map(|hv| hv.to_str().ok().map(str::to_string))
```

Comment:
As much as I love the cleanups, maybe we should all get better at cleaning them in seperate PR's so it's clearer to review. I'm as much guilty, so I need to improve too :)